### ISSUE TITLE
Xwt for Android

### ISSUE THREAD

**Issue by lytico** [mono/xwt#406]
I have started an android-backend:
https://github.com/lytico/xwt-mobile

Maybe some are interested to contribute.

**Comment by sevoku** [comment]
Wouldn't it make more sense to go with Xamarin (and Xamarin.Forms), or is MonoDroid still alive?

**Comment by DavidKarlas** [comment]
Imho... even desktop UIs are so different that is hard to make Xwt work nicely... Adding mobile into mix sounds like recipe for disaster... Maybe I'm wrong... Eto.Forms are working on this... But imo... Using PCL for business logic/models and 2 more libraries on top of that one with Xwt for desktop and Xamarin.Forms for mobile makes much more sense...

**Comment by sevoku** [comment]
Yes, even having desktop and mobile support in one toolkit, requires different implementations (I've seen it in Eto and its far from being ready). The only interesting thing would be to be able to use Xwt widgets with Xamarin.Forms. Imagine you have developed some kind of custom widget and you could reuse the code on a mobile platform :+1:

**Comment by DavidKarlas** [comment]
I really don't see much connection between this 2 UIs... on desktop you have at least 10 controls on screen with 3-5 in same row... On mobile you have max 10 controls and max 2 in same row... It's totally different UI. I can't even imagine scenario other then Login screen that makes sense...

**Comment by sevoku** [comment]
The first and best example is drawing. It must not be a complete control, or even a form (you're right it really doesn't make sense in most cases). But think of things like plotting graphs and such.

**Comment by lytico** [comment]
Indeed, I use it for the drawing part of my application. 
Currently I don't plan to implement the widget part of xwt. 
I don't think that the same xwt app could run on desktop and mobile, but most of the controls are very similar in both ui's.
One part missing in xwt is touch support, too. But implementation of this could be usefull on desktop too.

**Comment by sevoku** [comment]
Touch support is something I'll need in near future. I've looked at Wpf and Gtk APIs and they are completely different. And Gtk# does not include the Gestures classes, so a custom native wrapper is required. No idea about Mac right now. The first step would be to define some Xwt API for touch handling, which covers all possible toolkit patterns.

Back to topic: I'm not really sure right now, if it could really make sense to implement the complete widget part (Sure, GUIs are completely different, but having one Toolkit for all of them could still be nice). Its more a philosophical question, I think. If someone makes a complete Droid or Xamarin.Forms backend and requests a pull, why not?.
But drawing is really a good idea. Maybe we could add some backend marker (ToolkitFeatures.FullFeatured, ToolkitFeatures.Canvas ?), to indicate that the backend supports custom widgets only and add a note to the main Readme (like full featured backends: Gtk, Wpf, Mac; drawing backends: MonoDroid).

**Comment by lytico** [comment]
seems that ms is just doing that with universal apps
so its not so absurd thinking of universal xwt

**Comment by sevoku** [comment]
As I understand it is not really the same and it looks more like a media hype to me atm. We'll see how it will really look like soon, hopefully...
But after many months of working with xwt I see more and more requirements here. We have developed many custom widgets with Xwt for e.g. visualization of automation processes and other things. So when we'll proceed with mobile support, well have to decide to work on xwt to reuse our code or to use some web techniques and to rewrite everything from scratch. With my xwt experience it would be more efficient to create mobile xwt backends and to reuse our existing code.
For sure the guis would have a completely different layout but we would have to rewrite only the layout parts and not everything else.
But atm I can't make any decision for me cause the xwt development seems to stuck a little bit. Mobile backends would be really challenging, but only if the project has a real future...

**Comment by lytico** [comment]
it seems that xamarin.forms catches up ... they go desktop:

https://blog.xamarin.com/preview-bringing-macos-to-xamarin-forms/
